tests/MongoDB.Driver.Tests/Specifications/change-streams/ChangeStreamTestRunner.cs

```diff
@@ -0,0 +1,414 @@
+                RequireServer.Check().ClusterTypes(clusterTypes);
+            }
+
+            _databaseName = shared["database_name"].AsString;
+            _database2Name = shared["database2_name"].AsString;
+            _collectionName = shared["collection_name"].AsString;
+            _collection2Name = shared["collection2_name"].AsString;
+
+            CreateCollections();
+
+            List<ChangeStreamDocument<BsonDocument>> actualResult = null;
+            Exception actualException = null;
+            List<CommandStartedEvent> actualEvents = null;
+
+            var eventCapturer = CreateEventCapturer();
+            using (var client = CreateDisposableClient(eventCapturer))
+            {
+                try
+                {
+                    using (var cursor = Watch(client, test))
+                    {
+                        var globalClient = DriverTestConfiguration.Client;
+                        ExecuteOperations(globalClient, test["operations"].AsBsonArray);
+
+                        actualResult = ReadChangeStreamDocuments(cursor, test);
+                        actualEvents = GetEvents(eventCapturer);
+                    }
+                }
+                catch (Exception exception)
+                {
+                    actualException = exception;
+                }
+            }
+
+            if (test.Contains("expectations") && actualEvents != null)
+            {
+                var expectedEvents = test["expectations"].AsBsonArray.Cast<BsonDocument>().ToList();
+                AssertEvents(actualEvents, expectedEvents);
+            }
+
+            if (test.Contains("result"))
+            {
+                var expectedResult = test["result"].AsBsonDocument;
+                AssertResult(actualResult, actualException, expectedResult);
+            }
+        }
+
+        // private methods
+        private ClusterType[] MapTopologyToClusterTypes(BsonArray topologies)
+        {
+            var clusterTypes = new List<ClusterType>();
+            foreach (var topology in topologies.Select(i => i.AsString))
+            {
+                switch (topology)
+                {
+                    case "single": clusterTypes.Add(ClusterType.Standalone); break;
+                    case "replicaset": clusterTypes.Add(ClusterType.ReplicaSet); break;
+                    case "sharded": clusterTypes.Add(ClusterType.Sharded); break;
+                    default: throw new FormatException($"Invalid topology: \"{topology}\".");
+                }
+            }
+            return clusterTypes.ToArray();
+        }
+
+        private ChangeStreamOptions ParseChangeStreamOptions(BsonDocument document)
+        {
+            var options = new ChangeStreamOptions();
+
+            foreach (var element in document)
+            {
+                throw new FormatException($"Invalid change stream option: \"{element.Name}\".");
```

Comment:
This looks odd (for now) only because at the moment all change stream options in the JSON files are empty documents so I don't know what element names are valid.